src/main/java/com/codeit/sb01_deokhugam/domain/comment/controller/CommentController.java

```diff
@@ -1,4 +1,59 @@
 package com.codeit.sb01_deokhugam.domain.comment.controller;
 
+import com.codeit.sb01_deokhugam.domain.comment.dto.CommentCreateRequest;
+import com.codeit.sb01_deokhugam.domain.comment.dto.CommentDto;
+import com.codeit.sb01_deokhugam.domain.comment.dto.CommentUpdateRequest;
+import com.codeit.sb01_deokhugam.domain.comment.service.CommentService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/api/comments")
+@RequiredArgsConstructor
 public class CommentController {
+
+    private final CommentService commentService;
+
+    @PostMapping
+    public CommentDto create(@RequestBody CommentCreateRequest request,
+                             @RequestHeader("UserID") UUID userId,
+                             @RequestHeader("UserNickname") String userNickname) {
+        return commentService.create(request.reviewId(), userId, request.content(), userNickname);
+    }
+
+    @GetMapping("/review/{reviewId}")
+    public List<CommentDto> getComments(@PathVariable UUID reviewId,
+                                        @RequestParam(required = false) Instant after,
+                                        @RequestHeader("UserNickname") String userNickname) {
+        return commentService.getComments(reviewId, after, userNickname);
+    }
+
+    @GetMapping("/{commentId}")
+    public CommentDto getComment(@PathVariable UUID commentId,
+                                 @RequestHeader("UserNickname") String userNickname) {
+        return commentService.getCommentById(commentId, userNickname);
+    }
+
+    @PatchMapping("/{commentId}")
+    public CommentDto update(@PathVariable UUID commentId,
+                             @RequestHeader("UserID") UUID userId,
+                             @RequestBody CommentUpdateRequest request,
+                             @RequestHeader("UserNickname") String userNickname) {
+        return commentService.update(commentId, userId, request.content(), userNickname);
+    }
+
+    @DeleteMapping("/{commentId}")
+    public void delete(@PathVariable UUID commentId,
+                       @RequestHeader("UserID") UUID userId,
+                       @RequestParam(defaultValue = "false") boolean hard) {
+        if (hard) {
+            commentService.hardDelete(commentId);
+        } else {
+            commentService.softDelete(commentId, userId);
+        }
+    }
```

Comment:
제공된 프론트엔드 코드에서는 논리 삭제와 물리 삭제의 API를 구분해서 작성되어 있어요!!
제공된 swagger 문서를 한 번 확인해보시는게 좋을 것 같아요 😄 

개인 적으로 정답은 아니지만, MVC 구조에서 `Controller` 레이어에서는 서비스 로직을 넣어두지 않는 편이 역할 분리에 맞다고 생각해요.
```suggestion
    @DeleteMapping("/{commentId}")
    public void delete(@PathVariable UUID commentId,
                       @RequestHeader("UserID") UUID userId,
                       @RequestParam(defaultValue = "false") boolean hard) {
            commentService.hardDelete(commentId);
    }
```